### PR TITLE
Add properties and methods for animated image in Image class.

### PR DESCRIPTION
**Pull request by shmin00**
### Description of Change ###

Added properties and methods for playing animated image.

### Bugs Fixed ###

N/A

### API Changes ###

Added:
 - bool IsAnimated {get; } 
Get a value that indicates whether an image is animated.
 - bool IsPlaying {get; } 
Gets a value that indicated whether an animated image is playing.
 - void Play ();
Starts the animation of an animated image.
 - void Stop ();
Stops the animation of an anmated image.

### Behavioral Changes ###

None

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

### CODE REVIEW

File: Xamarin.Forms.Core/Image.cs [hunk 1]
```diff
@@ -169,5 +184,16 @@ void IImageController.SetIsLoading(bool isLoading)
...
+        public void Play()
+        {
+            if(IsAnimated)
+                IsPlaying = true;
+        }
+
+        public void Stop()
+        {
+            IsPlaying = false;
+        }
```
Comment (pauldipietro) [inline Xamarin.Forms.Core/Image.cs]: Just noting that the tabbing was incorrectly set here and above.

**Comment by StephaneDelcroix** [comment]
Can't we leave `Image` as is and implement this in an ImageSource ?

**Comment by shmin00** [comment]
@StephaneDelcroix,
I think that these properties and methods can be used after loading of the image and it is related to how the image displayed. 
Could I ask why you think that ImageSource is more appropriate than Image?

**Comment by StephaneDelcroix** [comment]
Hi @shmin00 that was a wild guess, but you're right, we do not need a new ImageSource. On the other hand, we don't want Play/Stop on Image for non-animated ones.

Does subclassing Image to AnimatedImage makes sense to you ?

**Comment by shmin00** [comment]
Hi @StephaneDelcroix, Thanks for you opinion. 
Yes, Play/Stop methods may not seem necessary for non-animated images.
AnimatedImage looks better for me. According to your suggestion, I'll make new patch to add AnimatedImage class which inherits Image class.
Above APIs will be changed to like below.

class AnimatedImage : Image
{
	+ bool IsPlaying()
	+ void Play()
	+ void Stop()
}
(IsAnimated property will be removed)
About the behavior, If a app developer wants to play animated image, he should use AnimatedImage class to load image. If he uses Image class, the image will be displayed same as non-animated image. 

During preparing the patch, I'd like to hear more opinion about this design.

**Comment by opcodewriter** [comment]
@jassmith I don't get it. Why did you close this one? Someone spent time on this PR, why not at least give some reasoning behind closing the issue?

**Comment by jassmith** [comment]
We're working with them on this in private :) No worries. Sorry the external appearance is worse than the internal reality.

**Comment by opcodewriter** [comment]
private? an animated image of a class is a secret? not sure if you're really joking
it's a simple functionality someone implemented, it doesn't make sense, why close the PR? PR exist for a reason..

**Comment by jassmith** [comment]
Feel free to re-open it if you want. The reason it was closed for now is it still needs work by the developer to work on iOS/UWP/WinRT/WinPhone. Since this work will not be done soon we closed it for now.

**Comment by opcodewriter** [comment]
The last comment is the guy asking an opinion from you on the class design so he can modify his PR. Closing PR means to me "we're not interested".
As a developer I have to choose which technology to use, so I'm watching what's happening with Xamarin Forms. I'm just trying to understand Xamarin's intentions with Xamarin Forms, and to me it's all blurry now.

**Comment by opcodewriter** [comment]
Sorry, I have no intentions to be harsh, I just want to understand what is going on

**Pull request closed without merging**